src/cosmicqc/cli.py

```diff
@@ -0,0 +1,111 @@
+"""
+Setup coSMicQC CLI through python-fire
+"""
+
+import inspect
+import sys
+import types
+from typing import Optional
+
+import fire
+from fire import helptext, inspectutils, value_types
+from fire.core import Display, FireError, _DictAsString, _OneLineResult
+from fire.trace import FireTrace
+
+from . import analyze
+
+
+# referenced from https://github.com/google/python-fire/pull/446
+# to be removed after python-fire merges changes (uncertain of timeline)
+def HasCustomRepr(component: object) -> bool:
+    """Reproduces above HasCustomStr function to determine if component has a
+    custom __repr__ method.
+
+    ...
+
+    Args:
+      component: The object to check for a custom __repr__ method.
+    Returns:
+      Whether `component` has a custom __repr__ method.
+    """
+    if hasattr(component, "__repr__"):
+        class_attrs = inspectutils.GetClassAttrsDict(type(component)) or {}
+        repr_attr = class_attrs.get("__repr__")
+        if repr_attr and repr_attr.defining_class is not object:
+            return True
+    return False
+
+
+# referenced with modifications from https://github.com/google/python-fire/pull/446
+# to be removed after python-fire merges changes (uncertain of timeline)
+# ruff: noqa: C901
+def _PrintResult(
+    component_trace: FireTrace, verbose: bool = False, serialize: Optional[bool] = None
+) -> None:
+    """Prints the result of the Fire call to stdout in a human readable way."""
```

Comment:
Recommend including args in this doc string to clarify what each of the parameters are.

Reply:
Thanks for this comment - I'm afraid of documenting this differently than what's found within `python-fire` as this might change and we only need this for a CLI-focused fix from the context of `coSMicQC`. I'm going to postpone updating this for now.